bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/schema/SchemaRegistry.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2021-2023 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.binding.tuya.internal.schema;
+
+import java.lang.reflect.Type;
+import java.util.List;
+
+import org.openhab.core.storage.Storage;
+import org.openhab.core.storage.StorageService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.smarthomej.binding.tuya.internal.util.SchemaDp;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * Registry for device channels
+ *
+ * @author Vitalii Herhel - Initial contribution
+ */
+@Component(service = SchemaRegistry.class, configurationPid = "tuya.schemaService")
+public class SchemaRegistry {
```

Comment:
What is the benefit over using the `Storage` directly?

Reply:
This change is not mandatory. It's an attempt to extract a clear interface for schema storage. Before that, we operated on storage directly and serialised/deserialised objects while interacting with it. Now, we see what type of objects can be stored here and we don't need to bridge Gson object along with the Storage.